xtable-core/src/main/java/org/apache/xtable/parquet/ParquetConversionSource.java

```diff
@@ -0,0 +1,193 @@
+                .name(tableName)
+                .layoutStrategy(dataLayoutStrategy)
+                .partitioningFields(partitionFields)
+                .readSchema(schema)
+                .latestCommitTime(Instant.ofEpochMilli(latestFile.getModificationTime()))
+                .build();
+    }
+
+    public List<InternalDataFile> getInternalDataFiles() {
+        List<InternalDataFile> internalDataFiles = null;
+        List<LocatedFileStatus> parquetFiles = getParquetFiles(hadoopConf, basePath);
+        InternalTable table = getTable(-1L);
+        internalDataFiles =
+                parquetFiles.stream()
+                        .map(
+                                file ->
+                                        InternalDataFile.builder()
+                                                .physicalPath(file.getPath().toString())
+                                                .fileFormat(FileFormat.APACHE_PARQUET)
+                                                .fileSizeBytes(file.getLen())
+                                                .partitionValues(partitionValueExtractor.extractPartitionValues(
+                                                        partitionValueExtractor.extractParquertPartitions(parquetMetadataExtractor.readParquetMetadata(
+                                                                hadoopConf, file.getPath()), file.getPath().toString()),
+                                                        basePath))
+                                                .lastModified(file.getModificationTime())
+                                                .columnStats(
+                                                        parquetStatsExtractor.getColumnStatsForaFile(
+                                                                parquetMetadataExtractor.readParquetMetadata(
+                                                                        hadoopConf, file.getPath())))
+                                                .build())
+                        .collect(Collectors.toList());
+        return internalDataFiles;
+    }
+
+    // since we are considering files instead of tables in parquet
+    @Override
+    public CommitsBacklog<java.lang.Long> getCommitsBacklog(
+            InstantsForIncrementalSync lastSyncInstant) {
+        long epochMilli = lastSyncInstant.getLastSyncInstant().toEpochMilli();
+        return null;
+    }
+
+    @Override
+    public TableChange getTableChangeForCommit(java.lang.Long commit) {
+        return null;
+    }
+
+    @Override
+    public InternalTable getCurrentTable() {
+        return null;
+    }
+
+
+    /**
+     * Here to get current snapshot listing all files hence the -1 is being passed
+     *
+     * @return
+     */
+    @Override
+    public InternalSnapshot getCurrentSnapshot() {
+        List<InternalDataFile> internalDataFiles = getInternalDataFiles();
+        InternalTable table = getTable(-1L);
+        return InternalSnapshot.builder()
```

Comment:
We'll need to set the `version` here. I am guessing it should be the last modification time but I need to think it through more.